OJS.Workers.ExecutionStrategies/NodeJs/NodeJsProjectRunJavaScriptProjectAndTestsWithPlaywrigth.cs

```diff
@@ -0,0 +1,279 @@
+    capture_output=True
+)
+
+print(process.stdout)
+
+executor.stop()
+
+";
+
+        protected string NginxFileContent => $@"
+worker_processes  1;
+
+events {{
+    worker_connections  1024;
+}}
+
+http {{
+    include mime.types;
+    types
+    {{
+        application/javascript mjs;
+    }}
+
+    default_type  application/octet-stream;
+    sendfile        on;
+    keepalive_timeout  65;
+
+    server {{
+      root /usr/share/nginx/html;
+
+      listen 80;
+      server_name localhost;
+
+      location / {{
+        try_files $uri $uri/ /index.html;
+        autoindex on;
+      }}
+    }}
+}}";
+
+        protected override IExecutionResult<TestResult> ExecuteAgainstTestsInput(
+            IExecutionContext<TestsInputModel> executionContext,
+            IExecutionResult<TestResult> result)
+        {
+            try
+            {
+                this.ExtractSubmissionFiles(executionContext);
+            }
+            catch (ArgumentException exception)
+            {
+                result.IsCompiledSuccessfully = false;
+                result.CompilerComment = exception.Message;
+
+                return result;
+            }
+
+            this.SaveTestsToFiles(executionContext.Input.Tests);
+            this.SaveNginxFile();
+
+            var codeSavePath = this.SavePythonCodeTemplateToTempFile();
+            var executor = this.CreateExecutor();
+            var checker = executionContext.Input.GetChecker();
+            return this.RunTests(codeSavePath, executor, checker, executionContext, result);
+        }
+
+        protected override IExecutionResult<TestResult> RunTests(
+            string codeSavePath,
+            IExecutor executor,
+            IChecker checker,
+            IExecutionContext<TestsInputModel> executionContext,
+            IExecutionResult<TestResult> result)
+        {
+            foreach (var test in executionContext.Input.Tests)
+            {
+                var processExecutionResult = this.Execute(executionContext, executor, codeSavePath, test.Input);
+                var testResults = this.ExtractTestResultsFromReceivedOutput(processExecutionResult, test.Id);
+                result.Results.AddRange(testResults);
+            }
+
+            return result;
+        }
+
+        protected override IExecutor CreateExecutor() => this.CreateExecutor(ProcessExecutorType.Standard);
+
+        private void ExtractSubmissionFiles<TInput>(IExecutionContext<TInput> executionContext)
+        {
+            
+            var trimmedAllowedFileExtensions = executionContext.AllowedFileExtensions?.Trim();
+            var allowedFileExtensions = (!trimmedAllowedFileExtensions?.StartsWith(".") ?? false)
+                ? $".{trimmedAllowedFileExtensions}"
+                : trimmedAllowedFileExtensions;
+
+            if (allowedFileExtensions != Constants.ZipFileExtension)
+            {
+                throw new ArgumentException("Submission file is not a zip file!");
+            }
```

Comment:
This check can be extracted into a method, i.e. `ValidateAllowedFileExtension()`